homeassistant/components/overkiz/climate_entities/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,203 @@
+
+OVERKIZ_TO_HVAC_ACTION = {
+    OverkizCommandParam.COOLING: HVACAction.COOLING,
+    OverkizCommandParam.HEATING: HVACAction.HEATING,
+}
+
+MAP_PRESET_TEMPERATURES = {
+    PRESET_COMFORT: OverkizState.CORE_COMFORT_ROOM_TEMPERATURE,
+    PRESET_ECO: OverkizState.CORE_ECO_ROOM_TEMPERATURE,
+    PRESET_AWAY: OverkizState.CORE_SECURED_POSITION_TEMPERATURE,
+}
+
+MODE_COMMAND_MAPPING = {
+    OverkizCommandParam.COMFORT: OverkizCommand.SET_COMFORT_TEMPERATURE,
+    OverkizCommandParam.ECO: OverkizCommand.SET_ECO_TEMPERATURE,
+    OverkizCommandParam.SECURED: OverkizCommand.SET_SECURED_POSITION_TEMPERATURE,
+}
+
+TEMPERATURE_SENSOR_DEVICE_INDEX = 2
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface.
+
+    The thermostat has 3 ways of working:
+      - Auto: Switch to eco/comfort temperature on a schedule (day/hour of the day)
+      - Manual comfort: The thermostat use the temperature of the comfort setting (19°C degree by default)
+      - Manual eco: The thermostat use the temperature of the eco setting (17°C by default)
+      - Freeze protection: The thermostat use the temperature of the freeze protection (7°C by default)
+
+    There's also the possibility to change the working mode, this can be used to change from a heated
+    floor to a cooling floor in the summer.
+    """
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    # Both min and max temp values have been retrieved from the Somfy Application.
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            TEMPERATURE_SENSOR_DEVICE_INDEX
+        )
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation i.e. heat, cool mode."""
+        if (
+            cast(str, self.executor.select_state(OverkizState.CORE_ON_OFF))
+            == OverkizCommandParam.OFF
+        ):
+            return HVACMode.OFF
+
+        state = cast(
+            str,
+            self.executor.select_state(
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_ACTIVE_MODE
+            ),
+        )
+        if mode := OVERKIZ_TO_HVAC_MODES[OverkizCommandParam(state)]:
+            return mode
+
+        if state is not None:
+            # Unknown and potentially a new state, log to make it easier to report
+            _LOGGER.warning(
+                "Overkiz %s state unknown: %s",
+                OverkizState.OVP_HEATING_TEMPERATURE_INTERFACE_ACTIVE_MODE,
+                state,
+            )
```

Comment:
Do we need this in core? I would say that we now know which states are there; and otherwise it will be in the log.

Reply:
If we do add it, we must also in other core Overkiz entities, which do not have that at the moment